### PR TITLE
CORDA-2619: Backwards compatibility: put c-i back on the node classpath by default.

### PR DESCRIPTION
**Pull request by mikehearn**
Also fixes some deprecation warnings whilst I'm here.

This will leave the handler open, but we need to revisit how we close the old handlers in future releases, as our current approach is too simplistic and can break under rolling upgrade.

### CODE REVIEW

**Comment by mikehearn** [comment]
@shamsasari Reverted the changes and tested with DemoBench + my "upgrade one node to c4" test. Seems to work. Let's see what CI makes of it.

**Comment by mikehearn** [comment]
There are two sample apps in the samples repo that would need adjustment as well: obligation-cordapp and whistleblower

**Comment by shamsasari** [comment]
In V3 the SwapIdentitiesHandler was registered manually (https://github.com/corda/corda/blob/df343379d2acd6bd8ab9856435c2c9c0783ea07c/node/src/main/kotlin/net/corda/node/internal/AbstractNode.kt#L604) because it didn't have the `InitiatedBy` annotation. It does now in V4, but I'm not sure if the node scans locally for initiating-intiatedBy pairs like it does for apps.

**Comment by mikehearn** [comment]
What's the best way to test this? I've been using the CashPaymentFlow and sending cash back and forth with "anonymous: true". I've put back the handler registration.

**Comment by shamsasari** [comment]
Sounds about right, as long as the C-I jar is not in `cordapps` dir.

**Comment by mikehearn** [comment]
OK, so it turns out the node is scanning itself, so the second registration was failing. Trying again ...

File: confidential-identities/src/main/kotlin/net/corda/confidential/SwapIdentitiesFlow.kt [hunk 1]
```diff
@@ -128,7 +128,7 @@ open class SwapIdentitiesException @JvmOverloads constructor(message: String, ca
 
 // This only exists for backwards compatibility
 @InitiatedBy(SwapIdentitiesFlow::class)
-private class SwapIdentitiesHandler(private val otherSide: FlowSession) : FlowLogic<Unit>() {
+class SwapIdentitiesHandler(private val otherSide: FlowSession) : FlowLogic<Unit>() {
```
Comment (shamsasari) [inline confidential-identities/src/main/kotlin/net/corda/confidential/SwapIdentitiesFlow.kt]: Why is this public now? Perhaps move this class into the node?

**Pull request merged**